File: Vector/OOPs/Day 2 Applying SOLID Principles, DRY, and YAGNI in Java-JS/BASIC/Part 2 Open_close principle (OCP)/Main.js
import Employee from './Employee.js';
import StandardBonusCalculator from './StanderdBonusCalculator.js';
import PerformanceBonusCalculator from './PerformanceBonusCalculator.js';

const standardEmployee = new Employee("Alice", 50000, new StandardBonusCalculator());
console.log("Standard Bonus: " + standardEmployee.calculateBonus());

const performanceEmployee = new Employee("Bob", 50000, new PerformanceBonusCalculator());
console.log("Performance Bonus: " + performanceEmployee.calculateBonus());

File: Vector/OOPs/Day 2 Applying SOLID Principles, DRY, and YAGNI in Java-JS/BASIC/Part 2 Open_close principle (OCP)/Main.java
public class Main {
    public static void main(String[] args) {
        Employee standardEmployee = new Employee("Alice", 50000, new StandardBonusCalculator());
        System.out.println("Standard Bonus: " + standardEmployee.calculateBonus());

        Employee performanceEmployee = new Employee("Bob", 50000, new PerformanceBonusCalculator());
        System.out.println("Performance Bonus: " + performanceEmployee.calculateBonus());
    }
}
